polkadot/node/core/pvf/common/src/executor_interface.rs

```diff
@@ -169,8 +172,8 @@ pub fn params_to_wasmtime_semantics(par: &ExecutorParams) -> Semantics {
 			ExecutorParam::PvfExecTimeout(_, _) => (), /* Not used here */
 		}
 	}
-	sem.deterministic_stack_limit = Some(stack_limit);
-	sem
+	sem.deterministic_stack_limit = Some(stack_limit.clone());
+	(sem, stack_limit)
```

Comment:
> You could clone it from semantics anytime from the caller's side as well

We were doing that, but it required a proof with non-local reasoning as it was behind an `Option`. As we are guaranteed to have the stack limit, I changed it so that the function signature itself provides the guarantee that it exists. Let me know if you have any ideas for refactoring it!

Reply:
Well, fair enough. I'm trying to invent something around `Infallible` result out of my head, but cannot give birth to it right away. Let's leave it for the next refactoring then :)